halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/kubernetes/KubernetesAccountValidator.java

```diff
@@ -111,6 +111,17 @@ private void validateKindConfig(ConfigProblemSetBuilder psBuilder, KubernetesAcc
       psBuilder.addProblem(ERROR, "At most one of \"kinds\" and \"omitKinds\" may be specified.");
     }
 
+    if (CollectionUtils.isNotEmpty(customResources)) {
+      List<String> kubernetesKindNotSet =
+          customResources.stream()
+              .map(KubernetesAccount.CustomKubernetesResource::getKubernetesKind)
+              .filter(cr -> (cr == null || cr.isEmpty()))
+              .collect(Collectors.toList());
+      if (CollectionUtils.isNotEmpty(kubernetesKindNotSet)) {
+        psBuilder.addProblem(ERROR, "Custom resources must have \"kubernetesKind\" set.");
```

Comment:
There's no indication to users on which is the `kubernetesKind` field. For example for a command like this:
```
hal config provider kubernetes account edit ACCOUNT --add-custom-resource "spinnakerKind:instances;versioned:true"
```
A better error message would be something like "missing custom resource name".